### PR TITLE
flag revokable clearances

### PR DESCRIPTION
**Pull request by ryan-semmler**
## Changes

- change the shape of the `/assignments/{campus_id}` response to include a boolean `"can_revoke"` field.
- remove the "allowed" top-level field in the json response
- changes to helper functions
    - move `get_authorization`
    - add `user_is_admin`
    - refactor and clean up endpoints using those two functions

## How was this tested?

manually and added pytests

## How were these changes documented?

n/a

## Is the version number updated?

yes

### CODE REVIEW

File: util/authorization.py [hunk 1]
```diff
@@ -6,9 +6,12 @@
 
 
 def get_authorization(authorization: str = Header(default=None)):
-    """
-    Extract authorization details out of the token
-    """
+    """Middleware to extract authorization details out of the token"""
     token = authorization.split(" ")[1]
```
Comment (JeremyGibson) [inline util/authorization.py:10]: Will the authorization always in every-case have something in [1] index?

**Review by JeremyGibson — APPROVED** [review]
This looks good, based on what I know about the process. Tests are appropriate.  I added a comment about the middleware, only because it is making a perhaps unfounded assumption about there always being a 1 index.  If there is even a remote chance that it could throw an exception, we should open a ticket to fix that. Since that is not part of this PR it does not need to be fixed here.

File: util/authorization.py (hunk 1 above)
Comment (ryan-semmler) [reply util/authorization.py:10]: Yes, without exception. The authorization always comes in as a string in the format `Bearer {some-token}`, this just cuts out the "Bearer" part

**Comment by ryan-semmler** [comment]
These changes were included in another PR and have been merged

**Pull request closed without merging**